pkg/sink/kafka/config.go

```diff
@@ -0,0 +1,145 @@
+	config.SetKey("compression.codec", compression)
+	if compression != "none" {
+		log.Info("Kafka producer uses " + compression + " compression algorithm")
+	}
+
+	// retrying may cause reordering unless enable.idempotence is set to true.
+	config.SetKey("retries", 0)
+	// The following configuration properties are adjusted automatically when idempotence is enabled:
+	// max.in.flight.requests.per.connection=5 (must be less than or equal to 5), retries=INT32_MAX (must be greater than 0), acks=all
+	// Producer instantation will fail if user-supplied configuration is incompatible.
+	// config.SetKey("enable.idempotence", true)
+	config.SetKey("max.in.flight", 1)
+	config.SetKey("request.required.acks", int(options.RequiredAcks))
+	config.SetKey("message.max.bytes", options.MaxMessageBytes)
+	config.SetKey("socket.timeout.ms", int(options.DialTimeout.Milliseconds()))
+	return config
+}
+
+func completeSASLConfig(config *kafka.ConfigMap, o *Options) error {
+	if o.SASL != nil && o.SASL.SASLMechanism != "" {
+		if o.EnableTLS {
+			_ = config.SetKey("security.protocol", ProtocolSASL_SSL)
+		} else {
+			_ = config.SetKey("security.protocol", ProtocolSASL)
+		}
+		_ = config.SetKey("sasl.mechanisms", o.SASL.SASLMechanism)
+		switch o.SASL.SASLMechanism {
+		case SASLTypeSCRAMSHA256, SASLTypeSCRAMSHA512, SASLTypePlaintext:
+			_ = config.SetKey("sasl.username", o.SASL.SASLUser)
+			_ = config.SetKey("sasl.password", o.SASL.SASLPassword)
+		case SASLTypeGSSAPI:
+			_ = config.SetKey("sasl.username", o.SASL.SASLUser)
+			_ = config.SetKey("sasl.kerberos.service.name", o.SASL.GSSAPI.ServiceName)
+			switch o.SASL.GSSAPI.AuthType {
+			case security.UserAuth:
+				_ = config.SetKey("sasl.password", o.SASL.SASLPassword)
+			case security.KeyTabAuth:
+				_ = config.SetKey("sasl.kerberos.keytab", o.SASL.GSSAPI.KeyTabPath)
+			}
+		case SASLTypeOAuth:
+			_ = config.SetKey("sasl.oauthbearer.token.endpoint.url", o.SASL.OAuth2.TokenURL)
+		}
+	}
+
+	return nil
+}
+
+func addVerifyPeerCertificate(config *kafka.ConfigMap, verifyCN []string) {
+	if len(verifyCN) != 0 {
+		checkCN := make(map[string]struct{})
+		for _, cn := range verifyCN {
+			cn = strings.TrimSpace(cn)
+			checkCN[cn] = struct{}{}
+		}
+		verifyPeerCertificate := func(
+			rawCerts [][]byte, verifiedChains [][]*x509.Certificate,
+		) error {
+			cns := make([]string, 0, len(verifiedChains))
+			for _, chains := range verifiedChains {
+				for _, chain := range chains {
+					cns = append(cns, chain.Subject.CommonName)
+					if _, match := checkCN[chain.Subject.CommonName]; match {
+						return nil
+					}
+				}
+			}
+			return errors.Errorf("client certificate authentication failed. "+
+				"The Common Name from the client certificate %v was not found "+
+				"in the configuration cluster-verify-cn with value: %s", cns, verifyCN)
+		}
+		_ = config.SetKey("ssl.certificate.verify_cb", verifyPeerCertificate)
+	}
+}
+
+func getLogLevel() int {
```

Comment:
How does this work, is print log by default?

Reply:
you could find `log_level` in https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md